moshi/src/main/java/com/squareup/moshi/ClassJsonAdapter.java

```diff
@@ -56,8 +56,12 @@
       if (rawType.isInterface() || rawType.isEnum()) return null;
       if (!annotations.isEmpty()) return null;
       if (Util.isPlatformType(rawType)) {
+        String messagePrefix = "Platform " + rawType;
+        if (type instanceof ParameterizedType) {
+          messagePrefix += " in " + type;
```

Comment:
Fun fact: this will produce three StringBuilders. No wait that's an unfun fact. The actual fun fact is that R8 1.6 (in AGP 3.6) will merge these into a single StringBuilder!

(Not that we care about three StringBuilders in an error branch anyway...)